### PR TITLE
bugfix/16568-data-options-merging

### LINKED ISSUE #16568: Global `data` options are not merged after using `Highcharts.setOptions`
#### Expected behaviour
Use `Highcharts.setOptions` with some options for `data`. Those should be merged with the chart options.

#### Actual behaviour
There is no merge from the `defaultOptions` in `data` options.

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/hv3geo4b/

#### Product version
9.3.0+

### PR DESCRIPTION
**Pull request by mfilipiec**
Fixed #16568, setting `data` options with `Highcharts.setOptions` did not work.

~~Fixed #16568, merging global data options with the chart options.~~

### CODE REVIEW

**Review by raf18seb — APPROVED** [review]
Good job!

**Review by raf18seb — CHANGES_REQUESTED** [review]
Tests are not passing, could you check this out?

File: samples/unit-tests/data/general/demo.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+const defaultOptions = { ...Highcharts.defaultOptions };
```
Comment (raf18seb) [inline samples/unit-tests/data/general/demo.js:1]: Unused variable, do we need this?

File: samples/unit-tests/data/general/demo.js [hunk 2]
```diff
@@ -242,6 +250,17 @@ QUnit.test('Data config on updates', function (assert) {
...
+    assert.strictEqual(
+        chart.series.length,
+        2,
+        'Global data options should be merged with the chart options (#16568).'
+    );
+    
+    // Back to default options after test.
+    Highcharts.setOptions({
+        data: null
+    });
```
Comment (raf18seb) [inline samples/unit-tests/data/general/demo.js:263]: It's good enough, but you can simply clear the `defaultOptions` this way:
```js
// Clear the default options for other tests
delete Highcharts.defaultOptions.data;
```

**Review by raf18seb — CHANGES_REQUESTED** [review]

**Review by raf18seb — APPROVED** [review]
Thanks!

**Review by KacperMadej — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Looks good, thanks!

**Pull request merged**